src/main/java/crazypants/enderio/material/OreDictionaryPreferences.java

```diff
@@ -62,7 +62,7 @@ public ItemStack getPreferred(ItemStack stack) {
     }
 
     if(result == null) {
-      result = stack;
+      result = stack.copy();
     }
     stackCache.put(key, result);
     return stack;
```

Comment:
Should not maybe this return stack.copy()? Then we solve the problem at the root.